src/energenieApi/energenieApi.ts

```diff
@@ -53,14 +54,14 @@ export class EnergenieApi {
   public toggleSocketPower(id: number, action: string): Promise<void> {
     const params = `params={ "id": ${id} }`;
 
-    return this.makeHttpRequest(() => this.httpClient.post(`${this.baseUrl}subdevices/` + action, params),
+    return this.makeHttpRequest(() => this.httpClient.post(`${getUrl('subdevices', this.baseUrl)}` + action, params),
```

Comment:
Nice, getUrl is very handy. Could you pull the calls out into variables i.e. `const url = getUrl('subdevices', this.baseUrl) + action;`. Just makes it a bit easier to read.

Reply:
You also don't need the string interpolation `${}` when you don't have a string literal.

Reply:
Whilst we're looking at this too, the "action" parameter should probably be removed from `platformAccessory` - the specific values required are an api return, we should decouple here.

Reply:
Great suggestions! Implemented.